users/views.py

```diff
@@ -156,18 +178,44 @@ def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
         context_data = super().get_context_data(**kwargs)
         context_data["year"] = datetime.datetime.now().year
         context_data["month"] = datetime.datetime.now().month
-        context_data["ordered_object_list"] = self._get_ordered_list_of_users()
+        context_data["active_users_list"] = self._get_ordered_list_of_users(self.queryset_active)
+        context_data["disabled_users_list"] = self._get_ordered_list_of_users(self.queryset_disabled)
+        context_data["UI_text"] = UsersListText
         return context_data
 
-    def _get_users_by_user_type(self, user_type: str) -> QuerySet:
-        return self.queryset.filter(user_type=user_type)
-
-    def _get_ordered_list_of_users(self) -> List[CustomUser]:
-        admins = self._get_users_by_user_type(CustomUser.UserType.ADMIN.name)
-        managers = self._get_users_by_user_type(CustomUser.UserType.MANAGER.name)
-        employees = self._get_users_by_user_type(CustomUser.UserType.EMPLOYEE.name)
+    def _get_ordered_list_of_users(self, queryset: QuerySet) -> List[CustomUser]:
+        admins = self._get_users_by_user_type(CustomUser.UserType.ADMIN.name, queryset)
+        managers = self._get_users_by_user_type(CustomUser.UserType.MANAGER.name, queryset)
+        employees = self._get_users_by_user_type(CustomUser.UserType.EMPLOYEE.name, queryset)
         return list(admins) + list(managers) + list(employees)
 
+    @staticmethod
+    def _get_users_by_user_type(user_type: str, queryset: QuerySet) -> QuerySet:
+        return queryset.filter(user_type=user_type)
+
+    def post(self, _request: HttpRequest) -> HttpResponseRedirectBase:
+        user_id_to_disable = self.request.POST.get("disable")
+        user_id_to_activate = self.request.POST.get("activate")
+
+        if user_id_to_disable is not None:
+            self._manage_user_status(user_id_to_disable)
+        elif user_id_to_activate is not None:
+            self._manage_user_status(user_id_to_activate)
+
+        return redirect("custom-users-list")
+
+    def _manage_user_status(self, user_id: int) -> None:
+        self._reverse_user_status(CustomUser.objects.get(id=user_id))
+
+    @staticmethod
+    def _reverse_user_status(user: CustomUser) -> CustomUser:
+        user_status = user.is_active
+
+        user.is_active = not user_status
```

Comment:
Why not just:
```python
user.is_active = not user.is_active
```
?

Reply:
Good question. Probably I've had a different concept here, but I decided to change approach. I've fixed that.